tests/dialects/test_mpi_lowering.py

```diff
@@ -0,0 +1,269 @@
+from xdsl.dialects import mpi, func, llvm, builtin
+from xdsl.ir import Operation, Attribute, OpResult, SSAValue
+from xdsl.irdl import irdl_op_definition, VarOpResult
+from xdsl.transforms import mpi_to_llvm
+
+info = mpi_to_llvm.MpiLibraryInfo()
+
+
+def extract_func_call(ops: list[Operation],
+                      name: str = 'MPI_') -> func.Call | None:
+    for op in ops:
+        if isinstance(op,
+                      func.Call) and op.callee.string_value().startswith(name):
+            return op
+
+
+def check_emitted_function_signature(ops: list[Operation],
+                                     name: str,
+                                     types: tuple[type[Attribute] | None, ...],
+                                     ignore_list: tuple[SSAValue] = tuple()):
+    call = extract_func_call(ops, name)
+    assert call is not None, f"Missing func.Call op to {name} in output!"
+    assert len(call.arguments) == len(types)
+    for i, type_ in enumerate(types):
+        arg = call.arguments[i]
+
+        # check that the argument type is correct (if constraint present)
+        if type_ is not None:
+            assert isinstance(
+                arg.typ, type_
+            ), f"Expected argument to be of type {type_} (got {arg.typ} instead)"
+
+        # check that all arguments originate in the emitted operations, except for the exceptions
+        if arg in ignore_list:
+            continue
+
+        assert isinstance(arg, OpResult)
+        assert arg.op in ops, f"Expected {arg.op} to be present in the emitted operations!"
+
+
+@irdl_op_definition
+class CreateTestValsOp(Operation):
+    name = "testing.test"
+    result: VarOpResult
+
+    @staticmethod
+    def get(*types: Attribute):
+        return CreateTestValsOp.build(result_types=[list(types)])
+
+
+def test_lower_mpi_init():
+    ops, result = mpi_to_llvm.LowerMpiInit(info).lower(mpi.Init.build())
```

Comment:
This is something that should be done in a filecheck test, IMO.

Reply:
Shouldn't we put most of these tests in filecheck tests?
Users will just use the `mpi-to-llvm` pass, so we should be able to cover it entirely with just these filecheck tests right?

Reply:
These tests check explicitly that the correct function signature is emitted by the rewrites. I feel that a filecheck is re-generated every time you check something, and that it might be easy to overlook a breaking change.

These tests are only here to explicitly enforce the MPI standard.